docs/sources/service.md

```diff
@@ -0,0 +1,108 @@
+# Ingress source
+
+The service source creates DNS entries based on `Service` resources.
+
+## Filtering the Services considered
+
+The `--service-type-filter` flag filters Service resources by their `spec.type`.
+The flag may be specified multiple times to allow multiple service types.
+
+This source supports the `--label-filter` flag, which filters Service resources
+by a set of labels.
+
+## Domain names
+
+The domain names of the DNS entries created from a Service are sourced from the following places:
+
+* Adds the domain names from any `external-dns.alpha.kubernetes.io/hostname` and/or
+`external-dns.alpha.kubernetes.io/internal-hostname` annotation.
+This behavior is suppressed if the `--ignore-hostname-annotation` flag was specified.
+
+* If no DNS entries were produced for a Service by the previous steps
```

Comment:
```suggestion
2. If no DNS entries were produced for a Service by the previous step
```

Reply:
This isn't the only place where items in the list depend on previous items not producing outputs. If we're going to take this sort of order-dependency as requiring a numbered list, we should do it consistently.